src/vnodeTransformers/util.ts

```diff
@@ -53,8 +53,12 @@ export const createVNodeTransformer = ({
       transformationCache.set(originalType, transformedType)
 
       registerStub({ source: originalType, stub: transformedType })
+      // https://github.com/vuejs/test-utils/issues/1829
+      // Teleport should return child nodes as a function
+      if (transformedType.name === 'teleport') {
```

Comment:
It might be better to check for the `Teleport` component instead of the name
```suggestion
      if (originalType === Teleport) {
```

Reply:
👍 but sadly this does not compile: I ended up borrowing the check from core: `const isTeleport = (type: any): boolean => type.__isTeleport`